common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -542,3 +544,171 @@ def test_courseware_a11y(self):
         self.courseware_page.a11y_audit.config.set_scope(
             include=['div.sequence-nav'])
         self.courseware_page.a11y_audit.check_for_accessibility_errors()
+
+
+@ddt.ddt
+class ProblemStateOnNavigationTest(UniqueCourseTest):
+    """
+    Test courseware with problems in multiple verticals
+    """
+    USERNAME = "STUDENT_TESTER"
+    EMAIL = "student101@example.com"
+
+    problem1_name = 'MULTIPLE CHOICE TEST PROBLEM 1'
+    problem2_name = 'MULTIPLE CHOICE TEST PROBLEM 2'
+
+    def setUp(self):
+        super(ProblemStateOnNavigationTest, self).setUp()
+
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+
+        # Install a course with section, tabs and multiple choice problems.
+        course_fix = CourseFixture(
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        )
+
+        course_fix.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section 1').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection 1,1').add_children(
+                    self.get_multiple_choice_problem(self.problem1_name),
+                    self.get_multiple_choice_problem(self.problem2_name),
+                ),
+            ),
+        ).install()
+
+        # Auto-auth register for the course.
+        AutoAuthPage(
+            self.browser, username=self.USERNAME, email=self.EMAIL,
+            course_id=self.course_id, staff=False
+        ).visit()
+
+        self.courseware_page.visit()
+        self.problem_page = ProblemPage(self.browser)
+
+    def get_multiple_choice_problem(self, problem_name):
+        """
+        Return the Multiple Choice Problem Descriptor, given the name of the problem.
+        """
+        factory = MultipleChoiceResponseXMLFactory()
+        xml_data = factory.build_xml(
+            question_text='The correct answer is Choice 2',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+
+        return XBlockFixtureDesc(
+            'problem',
+            problem_name,
+            data=xml_data,
+            metadata={'rerandomize': 'always'}
+        )
+
+    def go_to_tab_and_assert_problem(self, position, problem_name):
+        """
+        Go to sequential tab and assert that we are on problem whose name is given as a parameter.
+        Args:
+            position: Position of the sequential tab
+            problem_name: Name of the problem
+        """
+        self.courseware_page.go_to_sequential_position(position)
+        self.problem_page.wait_for_element_presence(
+            self.problem_page.CSS_PROBLEM_HEADER,
+            'wait for problem header'
+        )
+        self.assertEqual(self.problem_page.problem_name, problem_name)
+
+    @ddt.data(('choice_choice_1', 'label.choicegroup_incorrect'))
```

Comment:
I see your point. However, I don't think `ddt` is the way to do it. I'm personally ok with these hardcoded strings in the test, but if you have a very strong aesthetic aversion, you can make helper functions where you tuck away the strings.